src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1451,7 +1452,12 @@ public synchronized void restartLeaderElection(QuorumVerifier qvOLD, QuorumVerif
     }
 
     public String getNextDynamicConfigFilename() {
-        return configFilename + QuorumPeerConfig.nextDynamicConfigFileSuffix;
+        if (configFilename != null) {
```

Comment:
Thanks for review, Alex. I'll look into making configFileName not null for this test, which is more deterministic than a random id generator.